### PR TITLE
[Exchanges] add supports_websockets api

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/trading-backend/pull/4

### CODE REVIEW

File: octobot_trading/api/exchange.py [hunk 1]
```diff
@@ -149,6 +149,12 @@ def get_has_websocket(exchange_manager) -> bool:
     return exchange_manager.has_websocket
 
 
+def supports_websockets(exchange_name: str, tentacles_setup_config) -> bool:
+    exchange_manager = exchanges.ExchangeManager({}, exchange_name)
```
Comment (Herklos) [inline octobot_trading/api/exchange.py:153]: I don't think it's a good idea to create ExchangeManager instances to check if a list of exchanges supports websocket

Comment (GuillaumeDSM) [reply octobot_trading/api/exchange.py:153]: The problem is that we need to go there to check a websocket https://github.com/Drakkar-Software/OctoBot-Trading/blob/master/octobot_trading/exchanges/implementations/ccxt_websocket_exchange.py#L28 (or we can reimplement a new method but that will duplicate some code)

Comment (Herklos) [reply octobot_trading/api/exchange.py:153]: We should implement it differently

File: octobot_trading/exchanges/util/exchange_util.py [hunk 2]
```diff
@@ -76,14 +81,56 @@ def get_order_side(order_type):
...
+
+
+async def is_compatible_account(exchange_name: str, exchange_config: dict, tentacles_setup_config) -> (bool, str):
+    local_exchange_manager = exchange_manager.ExchangeManager(
+        _get_minimal_exchange_config(exchange_name, exchange_config),
+        exchange_name
+    )
+    local_exchange_manager.tentacles_setup_config = tentacles_setup_config
+    # Avoid exchange authentication error log
+    local_exchange_manager.is_collecting = True
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_util.py:121]: I think we should rename this attribute to fit this need. But for now it's ok

File: tests/exchanges/util/test_exchange_util.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+@pytest.fixture()
+def exchange_config():
+    return {
+        commons_constants.CONFIG_EXCHANGE_KEY: commons_configuration.encrypt("01234").decode(),
+        commons_constants.CONFIG_EXCHANGE_SECRET: commons_configuration.encrypt("012345").decode()
+    }
+
+
+async def test_is_compatible_account_with_checked_exchange(exchange_config, tentacles_setup_config):
```
Comment (Herklos) [inline tests/exchanges/util/test_exchange_util.py:43]: :+1:

File: tests/exchanges/util/test_exchange_util.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        is_valid_account_mock.assert_called_once()
+    with mock.patch.object(trading_backend.exchanges.Binance, "is_valid_account",
+                           mock.AsyncMock(return_value=(False, "plop"))) as is_valid_account_mock:
+        compatible, error = await exchanges.is_compatible_account("binance", exchange_config, tentacles_setup_config)
+        assert compatible is False
+        assert "plop" in error and len(error) > len("plop")
+        is_valid_account_mock.assert_called_once()
+
+
+async def test_is_compatible_account_with_unchecked_exchange(exchange_config, tentacles_setup_config):
```
Comment (Herklos) [inline tests/exchanges/util/test_exchange_util.py:58]: :+1:

File: tests/exchanges/util/test_exchange_util.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                           mock.AsyncMock(return_value=(True, None))) as is_valid_account_mock:
+        compatible, error = await exchanges.is_compatible_account("binance", exchange_config, tentacles_setup_config)
+        assert compatible is True
+        assert error is None
+        is_valid_account_mock.assert_called_once()
+    with mock.patch.object(trading_backend.exchanges.Binance, "is_valid_account",
+                           mock.AsyncMock(return_value=(False, "plop"))) as is_valid_account_mock:
+        compatible, error = await exchanges.is_compatible_account("binance", exchange_config, tentacles_setup_config)
+        assert compatible is False
+        assert "plop" in error and len(error) > len("plop")
```
Comment (Herklos) [inline tests/exchanges/util/test_exchange_util.py:54]: :smile:

**Review by Herklos — APPROVED** [review]

**Pull request merged**